### YamlMime:ManagedReference
items:
- uid: System.DuplicateWaitObjectException
  id: DuplicateWaitObjectException
  children:
  - System.DuplicateWaitObjectException.#ctor
  - System.DuplicateWaitObjectException.#ctor(System.String)
  - System.DuplicateWaitObjectException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.DuplicateWaitObjectException.#ctor(System.String,System.Exception)
  - System.DuplicateWaitObjectException.#ctor(System.String,System.String)
  langs:
  - csharp
  name: DuplicateWaitObjectException
  nameWithType: DuplicateWaitObjectException
  fullName: System.DuplicateWaitObjectException
  type: Class
  summary: "对象的同步对象数组中多次出现时，将引发异常。"
  remarks: "公共语言运行时提供了基于同步对象的数组中等待执行的线程同步机制<xref:System.Threading.WaitHandle>对象。</xref:System.Threading.WaitHandle> 如果数组<xref:System.Threading.WaitHandle>对象传递给<xref:System.Threading.WaitHandle.WaitAll%2A>或<xref:System.Threading.WaitHandle.WaitAny%2A>包含任何重复的操作系统句柄，则会引发 DuplicateWaitObjectException。</xref:System.Threading.WaitHandle.WaitAny%2A> </xref:System.Threading.WaitHandle.WaitAll%2A> </xref:System.Threading.WaitHandle> 有关详细信息，请参阅<xref:System.Threading.WaitHandle>。</xref:System.Threading.WaitHandle>       DuplicateWaitObjectException 使用 HRESULT COR_E_DUPLICATEWAITOBJECT，其值 0x80131529。       DuplicateWaitObjectException 的实例的初始属性值的列表，请参阅<xref:System.DuplicateWaitObjectException.%23ctor%2A>构造函数。</xref:System.DuplicateWaitObjectException.%23ctor%2A>      1> [!NOTE]&1;> 此异常未包含在[.NET for Windows Store 应用](http://go.microsoft.com/fwlink/?LinkID=247912)或的详细信息，但它将引发的某些成员。 若要在这种情况下捕获异常，编写`catch`语句<xref:System.ArgumentException>相反。</xref:System.ArgumentException>"
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class DuplicateWaitObjectException : ArgumentException
  inheritance:
  - System.ArgumentException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.DuplicateWaitObjectException.#ctor
  id: '#ctor'
  parent: System.DuplicateWaitObjectException
  langs:
  - csharp
  name: DuplicateWaitObjectException()
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException()
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "初始化的新实例<xref href=&quot;System.DuplicateWaitObjectException&quot;></xref>类。"
  remarks: "此构造函数初始化<xref:System.Exception.Message%2A>为系统提供的消息描述该错误，如&quot;重复自变量中的对象&quot;。 将新实例的属性</xref:System.Exception.Message%2A> 此消息将当前系统区域性考虑在内。       下表显示<xref:System.DuplicateWaitObjectException>.</xref:System.DuplicateWaitObjectException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|空引用 (`Nothing`在 Visual Basic 中)。 |  |<xref:System.ArgumentException.Message%2A>|本地化的错误消息字符串。 |</xref:System.ArgumentException.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public DuplicateWaitObjectException ();
    parameters: []
  overload: System.DuplicateWaitObjectException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.DuplicateWaitObjectException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.DuplicateWaitObjectException
  langs:
  - csharp
  name: DuplicateWaitObjectException(String)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(String)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "初始化的新实例<xref href=&quot;System.DuplicateWaitObjectException&quot;></xref>同名会导致此异常的参数的类。"
  remarks: "内容`parameterName`旨在为人所理解。       下表显示<xref:System.DuplicateWaitObjectException>.</xref:System.DuplicateWaitObjectException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|空引用 (`Nothing`在 Visual Basic 中)。 |  |<xref:System.ArgumentException.Message%2A>|本地化的错误消息字符串。 |  |<xref:System.ArgumentException.ParamName%2A>|参数名称字符串。 |</xref:System.ArgumentException.ParamName%2A> </xref:System.ArgumentException.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public DuplicateWaitObjectException (string parameterName);
    parameters:
    - id: parameterName
      type: System.String
      description: "导致异常的参数名称。"
  overload: System.DuplicateWaitObjectException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.DuplicateWaitObjectException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.DuplicateWaitObjectException
  langs:
  - csharp
  name: DuplicateWaitObjectException(SerializationInfo,StreamingContext)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(SerializationInfo,StreamingContext)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "初始化的新实例<xref href=&quot;System.DuplicateWaitObjectException&quot;></xref>用序列化数据的类。"
  remarks: "在反序列化，以便重建通过流传输的异常对象过程中调用此构造函数。 有关详细信息，请参阅[XML 和 SOAP 序列化](~/add/includes/ajax-current-ext-md.md)。"
  syntax:
    content: protected DuplicateWaitObjectException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "承载序列化的对象数据的对象。"
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "有关源或目标的上下文信息。"
  overload: System.DuplicateWaitObjectException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.DuplicateWaitObjectException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.DuplicateWaitObjectException
  langs:
  - csharp
  name: DuplicateWaitObjectException(String,Exception)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(String,Exception)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "初始化的新实例<xref href=&quot;System.DuplicateWaitObjectException&quot;></xref>使用指定的错误消息和对导致此异常的内部异常的引用的类。"
  remarks: "前一个异常的直接结果应包含中的前一个异常的引用引发的异常，<xref:System.Exception.InnerException%2A>属性。</xref:System.Exception.InnerException%2A> <xref:System.Exception.InnerException%2A>属性返回传递给此构造函数的相同值。</xref:System.Exception.InnerException%2A>       下表显示<xref:System.DuplicateWaitObjectException>.</xref:System.DuplicateWaitObjectException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|指定的异常`innerException`。 |  |<xref:System.ArgumentException.Message%2A>|指定的错误消息字符串`message`。 |</xref:System.ArgumentException.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public DuplicateWaitObjectException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "解释异常原因的错误消息。"
    - id: innerException
      type: System.Exception
      description: "导致当前异常的异常。 如果`innerException`参数不是<xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;></xref>中, 引发当前异常<xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;></xref>处理内部异常的块。"
  overload: System.DuplicateWaitObjectException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.DuplicateWaitObjectException.#ctor(System.String,System.String)
  id: '#ctor(System.String,System.String)'
  parent: System.DuplicateWaitObjectException
  langs:
  - csharp
  name: DuplicateWaitObjectException(String,String)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(String,String)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(String,String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "初始化的新实例<xref href=&quot;System.DuplicateWaitObjectException&quot;></xref>使用指定的错误消息和导致此异常的参数的名称的类。"
  remarks: "内容`parameterName`和`message`参数旨在为人所理解。 此构造函数的调用方需要确保这些字符串已针对当前系统区域性进行了本地化。       前一个异常的直接结果应包含中的前一个异常的引用引发的异常，<xref:System.Exception.InnerException%2A>属性。</xref:System.Exception.InnerException%2A> <xref:System.Exception.InnerException%2A>属性返回相同的值传递到构造函数中，则为空引用 (`Nothing`在 Visual Basic 中) 如果<xref:System.Exception.InnerException%2A>属性不会提供给构造函数的内部异常值。</xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       下表显示<xref:System.DuplicateWaitObjectException>.</xref:System.DuplicateWaitObjectException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|空引用 (`Nothing`在 Visual Basic 中) |  |<xref:System.ArgumentException.Message%2A>|错误消息字符串。 |  |<xref:System.ArgumentException.ParamName%2A>|参数名称字符串。 |</xref:System.ArgumentException.ParamName%2A> </xref:System.ArgumentException.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public DuplicateWaitObjectException (string parameterName, string message);
    parameters:
    - id: parameterName
      type: System.String
      description: "导致异常的参数名称。"
    - id: message
      type: System.String
      description: "描述错误的消息。"
  overload: System.DuplicateWaitObjectException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.ArgumentException
  isExternal: false
  name: System.ArgumentException
- uid: System.DuplicateWaitObjectException.#ctor
  parent: System.DuplicateWaitObjectException
  isExternal: false
  name: DuplicateWaitObjectException()
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException()
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException()
- uid: System.DuplicateWaitObjectException.#ctor(System.String)
  parent: System.DuplicateWaitObjectException
  isExternal: false
  name: DuplicateWaitObjectException(String)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(String)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.DuplicateWaitObjectException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.DuplicateWaitObjectException
  isExternal: false
  name: DuplicateWaitObjectException(SerializationInfo,StreamingContext)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(SerializationInfo,StreamingContext)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.DuplicateWaitObjectException.#ctor(System.String,System.Exception)
  parent: System.DuplicateWaitObjectException
  isExternal: false
  name: DuplicateWaitObjectException(String,Exception)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(String,Exception)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.DuplicateWaitObjectException.#ctor(System.String,System.String)
  parent: System.DuplicateWaitObjectException
  isExternal: false
  name: DuplicateWaitObjectException(String,String)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(String,String)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(String,String)
- uid: System.DuplicateWaitObjectException.#ctor*
  parent: System.DuplicateWaitObjectException
  isExternal: false
  name: DuplicateWaitObjectException
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException
